src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectKey.cs

```diff
@@ -13,7 +13,7 @@ namespace Microsoft.CodeAnalysis.Razor.ProjectSystem;
 /// A very light wrapper around a file path, used to ensure consistency across the code base for what constitutes the unique
 /// identifier for a project.
 /// </summary>
-internal sealed class ProjectKey : IEquatable<ProjectKey>
+internal readonly struct ProjectKey : IEquatable<ProjectKey>
```

Comment:
Can this be a record struct?

Reply:
I thought about it being a record when I originally wrote this, but in the end decided it wasn't worth it because we'd have to overwrite everything the record-ness gives us anyway, since we use a `FilePathComparer`. Also, I only want to allow static creation, so couldn't even save the property declaration by using a primary constructor.

I think those same arguments apply here. I don't really like that `new ProjectKey()` is possible 😛

Reply:
Alright, I have bowed to the weight of peer pressure and converted to a record struct. I remain skeptical that the 4 lines of code I removed were worth it 😛